Make Header Paths in Dictionary Relative

ChristianTackeGSI opened the pull request:
(I am working in the SDE group at GSI with @MohammadAlTurany and I am currently improving our packaging of vmc.)

According to [these root release notes](https://root.cern/doc/v618/release-notes.html#header-location-and-root_generate_dictionary-root_standard_library_package) the listed headers in `ROOT_GENERATE_DICTIONARY` should use relative paths.

Before this change, the headers are loaded from the source/build tree. If the source/build tree is removed after installation, the headers are not found any more.

With this change, the headers in the install tree are found by the interpreter, when `ROOT_INCLUDE_PATH` is set.

ChristianTackeGSI committed:
Make Header Paths in Dictionary Relative

According to

https://root.cern/doc/v618/release-notes.html#header-location-and-root_generate_dictionary-root_standard_library_package

the listed headers in ROOT_GENERATE_DICTIONARY should use
relative paths.

Before this change, the headers are loaded from the
source/build tree. If the source/build tree is removed
after installation, the headers are not found any more.

With this change, the headers in the install tree are found
by the interpreter, when ROOT_INCLUDE_PATH is set.

ihrivnac commented:
@ChristianTackeGSI , 
Thank you for this fix. There is already a related issue https://github.com/vmc-project/geant4_vmc/issues/13 where several solutions are proposed. You are welcome to contribute here with your opinion.

ChristianTackeGSI commented:
Thanks for the invitation! I will read up on that discussion and see, if I can contribute something useful.

The quick note: The proposed patch in this PR should not break anything (that wasn't already broekn). So it shouldn't hurt.

ChristianTackeGSI committed:
Change Include Path List in Dictionary

The generated dictionary used to contain the source/build
tree paths as the search path for include files. This works
for developers, but isn't good for the installed module.
In the worst case, it's a security risk, because the
source/build paths could be abused by an attacker.

So what is this change doing:
1. Append the install path to the search path. This is done
   using the OPTIONS argument to ROOT_GENERATE_DICTIONARY,
   which is appended at the end.
2. This doesn't affect the dictionary generation, because
   all headers should be findable in earlier (sourc/build)
   search path elements.
3. Use "-excludePath" to remove (nearly) all source/build
   path elements from the search path, that is finally
   stored in the dictionary.
4. This way, the install paths are first in the search list
   and all relevant headers should be findable there after
   installing.

Important Note:

For developers this means, that they either have to install
always to use things or use ROOT_INCLUDE_PATH to point to
the source/build tree.
Otherwise they might get outdated headers from the install
tree.

ChristianTackeGSI commented:
Added changes, as discussed in https://github.com/vmc-project/geant4_vmc/issues/13
